src/plone/restapi/tests/test_translations.py

```diff
@@ -206,3 +206,74 @@ def test_calling_with_an_untranslated_content_gives_400(self):
             json={"language": "es"},
         )
         self.assertEqual(400, response.status_code)
+
+
+@unittest.skipUnless(
+    PAM_INSTALLED, "plone.app.multilingual is installed by default only in Plone 5"
+)  # NOQA
+class TestCreateContentsAsTranslations(unittest.TestCase):
+    layer = PLONE_RESTAPI_DX_PAM_FUNCTIONAL_TESTING
+
+    def setUp(self):
+        self.portal = self.layer["portal"]
+        self.request = self.layer["request"]
+        alsoProvides(self.layer["request"], IPloneAppMultilingualInstalled)
+        login(self.portal, SITE_OWNER_NAME)
+        self.es_content = createContentInContainer(
+            self.portal["es"], "Document", title=u"Test document"
+        )
+        transaction.commit()
+
+    def test_post_to_folder_creates_document_translated(self):
+        response = requests.post(
+            "{}/de".format(self.portal.absolute_url()),
+            headers={"Accept": "application/json"},
+            auth=(SITE_OWNER_NAME, SITE_OWNER_PASSWORD),
+            json={
+                "@type": "Document",
+                "id": "mydocument",
+                "title": "My Document DE",
+                "translationOf": self.es_content.UID(),
+                "language": "de",
+            },
+        )
+        self.assertEqual(201, response.status_code)
+        transaction.commit()
+
+        manager = ITranslationManager(self.es_content)
+
+        self.assertTrue("de" in manager.get_translations())
+        self.assertEqual("My Document DE", manager.get_translations()["de"].title)
+
+        self.assertEqual("Document", response.json().get("@type"))
+        self.assertEqual("mydocument", response.json().get("id"))
+        self.assertEqual("My Document DE", response.json().get("title"))
+
+
+@unittest.skipUnless(
+    PAM_INSTALLED, "plone.app.multilingual is installed by default only in Plone 5"
+)  # NOQA
+class TestTranslationLocator(unittest.TestCase):
+    layer = PLONE_RESTAPI_DX_PAM_FUNCTIONAL_TESTING
+
+    def setUp(self):
+        self.portal = self.layer["portal"]
+        self.request = self.layer["request"]
+        alsoProvides(self.layer["request"], IPloneAppMultilingualInstalled)
+        login(self.portal, SITE_OWNER_NAME)
+        self.es_content = createContentInContainer(
+            self.portal["es"], "Document", title=u"Test document"
+        )
+        transaction.commit()
+
+    def test_translation_locator(self):
+        response = requests.get(
+            "{}/@translation-locator?targetLanguage=de".format(
+                self.es_content.absolute_url()
+            ),
+            headers={"Accept": "application/json"},
+            auth=(SITE_OWNER_NAME, SITE_OWNER_PASSWORD),
+        )
+        self.assertEqual(200, response.status_code)
+
+        self.assertEqual("http://localhost:55001/plone/de", response.json().get("@id"))
```

Comment:
@sneridagh you can not assume that the testing ZSERVER_PORT is always 55001. The jenkins.plone.org testrunner randomly assigns the ZSERVER_PORT. That's why the Plone coredev Jenkins jobs fail.

Reply:
Fixed.